deploy-queue/src/main.rs

```diff
@@ -0,0 +1,379 @@
+            match verify_deployment_can_be_finished(&db_client, *deployment_id).await {
+                Ok(true) => {
+                    // Update deployment record to set finish_timestamp
+                    match finish_deployment(&db_client, *deployment_id).await {
+                        Ok(()) => {
+                            log::info!("Successfully finished deployment with ID: {}", deployment_id);
+                        }
+                        Err(e) => {
+                            log::error!("Failed to update deployment record: {}", e);
+                            anyhow::bail!("Database update failed: {}", e);
+                        }
+                    }
+                }
+                Ok(false) => {
+                    anyhow::bail!("Deployment {} cannot be finished (not started or already finished/cancelled)", deployment_id);
+                }
+                Err(e) => {
+                    log::error!("Failed to verify deployment state: {}", e);
+                    anyhow::bail!("Database query failed: {}", e);
+                }
+            }
+        }
+        Mode::Cancel { deployment_id, cancellation_note } => {
+            log::info!("Cancelling deployment with ID: {}", deployment_id);
+
+            // Verify deployment exists and is in a valid state for cancellation
+            match verify_deployment_can_be_cancelled(&db_client, *deployment_id).await {
+                Ok(true) => {
+                    // Update deployment record to set cancellation_timestamp
+                    match cancel_deployment(&db_client, *deployment_id, cancellation_note.as_deref()).await {
+                        Ok(()) => {
+                            log::info!("Successfully cancelled deployment with ID: {}", deployment_id);
+                        }
+                        Err(e) => {
+                            log::error!("Failed to update deployment record: {}", e);
+                            anyhow::bail!("Database update failed: {}", e);
+                        }
+                    }
+                }
+                Ok(false) => {
+                    anyhow::bail!("Deployment {} cannot be cancelled (already finished or cancelled)", deployment_id);
+                }
+                Err(e) => {
+                    log::error!("Failed to verify deployment state: {}", e);
+                    anyhow::bail!("Database query failed: {}", e);
+                }
+            }
+        }
+    }
+
+    Ok(())
+}
+
+/// Create a database connection pool and return it
+async fn create_db_connection() -> Result<Pool<Postgres>, SqlxError> {
+    let database_url = std::env::var("DEPLOY_QUEUE_DATABASE_URL")
+        .map_err(|_| SqlxError::Configuration(
+            "DEPLOY_QUEUE_DATABASE_URL environment variable is required".into()
+        ))?;
+
+    let pool = PgPoolOptions::new()
+        .max_connections(5)
+        .acquire_timeout(Duration::from_secs(10))
```

Comment:
why deviate from the defaults here?

```suggestion
```

The defaults are 10 connections and 30 seconds timeout, which seem reasonable to me for the usecase here.